modules/REST/Abstract_Controller.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST;
+
+use WP_Error;
+use WP_REST_Controller;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Abstract REST controller.
+ *
+ * @since 3.7
+ */
+abstract class Abstract_Controller extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param int $version Optional. Version of the API. Default is `1`.
+	 *
+	 * @phpstan-param positive-int $version
+	 */
+	public function __construct( int $version = 1 ) {
+		$this->namespace = "pll/v{$version}";
+	}
```

Comment:
Since this is *all* V1, I suggest to remove the constructor for now.

Reply:
How do you suggest to set the property then?

Reply:
Directly in the class.

Reply:
Done in [12f1146](https://github.com/polylang/polylang/pull/1569/commits/12f11463a675f3797307b8a81b5d231e17de35a7).